datanommer.models/datanommer/models/__init__.py

```diff
@@ -128,7 +131,14 @@ def add(envelope):
     obj.msg = message['msg']
     obj.headers = headers
 
-    session.add(obj)
+    try:
+        session.add(obj)
+        session.flush()
```

Comment:
@jeremycline That is not consistent with the behavior I'm seeing. Remove the "try/except" and run the tests and you'll see an `IntegrityError` raised when `flush()` is called. I believe the doc you quoted there is explaining that `flush()` just issues the statements to the database, but does not issue any `commit` or `rollback`.